lib/closex/http_client.ex

```diff
@@ -89,15 +90,47 @@ defmodule Closex.HTTPClient do
 
   # Private stuff...
 
-  defp find(resource, search_term, opts) do
-    opts = merge_search_term_into_opts(search_term, opts)
-
+  defp find(resource, opts) do
     get("/#{resource}/", [], opts)
     |> handle_response
   end
 
+  # This function will attempt to carry out a `find` operation. If that fails
+  # with a 429 (Too Many Requests), close.io will provide us rate limiting
+  # information. Use this to wait till the end of the reset window and retry.
+  # See https://developer.close.io/#ratelimits for more info
+  #
+  # REVIEW: I've only implemented this on `find` as that's the only place we're
+  # hitting rate limits and it limits the scope of this change (in case of
+  # fubar). Should this be applied more generally?
+  defp find_respecting_rate_limit(resource, search_term, opts) do
+    opts = merge_search_term_into_opts(search_term, opts)
+
+    response = find(resource, opts)
+
+    case response do
+      {:error, %{status_code: 429, body: %{"rate_reset" => rate_reset}}} ->
+        wait_and_retry(resource, opts, rate_reset)
+
+      no_rate_limit ->
+        no_rate_limit
+    end
+  end
+
+  defp wait_and_retry(resource, opts, rate_reset) do
+    {seconds, _remainder} = Integer.parse(rate_reset)
+
+    @sleep_module.sleep(:timer.seconds(seconds + 1))
+
+    # REVIEW: This is purely for tests, is that a smell? Any alternatives?
```

Comment:
This is definitely a smell

Reply:
Yup! Any alternatives?

Reply:
It's not perfect, but now that it is a mock you can test the number of calls to the sleep module which basically does the same thing

Reply:
Ooh! that's a great point, I'll pull it into that!